yjit/src/stats.rs

```diff
@@ -839,3 +842,17 @@ fn global_allocation_size() -> usize {
     let stats = GLOBAL_ALLOCATOR.stats();
     stats.bytes_allocated.saturating_sub(stats.bytes_deallocated)
 }
+
+/// Measure the time taken by func() and add that to yjit_compile_time.
+pub fn with_compile_time<F, R>(func: F) -> R where F: FnOnce() -> R {
+    if get_option!(gen_stats) {
+        let start = SystemTime::now();
+        let ret = func();
+        let nanos = SystemTime::now().duration_since(start)
```

Comment:
I know very little of Rust, but looking at the doc this seem to use the "real time" system clock.

Generally for these things we use the monotonic clock, as it never decrease and is marginally faster.

Apparently you just need to use `Instant` instead ? https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/std/time/struct.Instant.html

Reply:
Thanks! https://github.com/ruby/ruby/pull/8417/commits/42b9e18d966fda0cb1ee1baae1ff02c8433b83df